packages/headless/src/controllers/core/facets/facet-dependencies-manager/headless-facet-dependencies-manager.ts

```diff
@@ -0,0 +1,146 @@
+import {CoreEngine} from '../../../../app/engine';
+import {facetOptions} from '../../../../app/reducers';
+import {
+  disableFacet,
+  enableFacet,
+} from '../../../../features/facet-options/facet-options-actions';
+import {AnyFacetValueRequest} from '../../../../features/facets/generic/interfaces/generic-facet-request';
+import {AnyFacetValue} from '../../../../features/facets/generic/interfaces/generic-facet-response';
+import {
+  CategoryFacetSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetSection,
+  NumericFacetSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+
+export interface AnyFacetDependency<T extends AnyFacetValueRequest> {
+  parentFacetId: string;
+  isDependencyMet: (parentValues: T[]) => boolean;
+}
+
+export interface FacetDependenciesManagerProps {
+  dependentFacetId: string;
+  dependencies: AnyFacetDependency<AnyFacetValueRequest>[];
+}
+
+export interface FacetDependenciesManager {
+  stopWatching(): void;
+}
+
+export function buildFacetDependenciesManager(
```

Comment:
> This is not a controller though, since it does not use `buildController`, does not return the same shape as a controller (for example: a `subscribe` function), even if it follows the `build` pattern, with the `engine` and `props`.
> 
> That's kind of misleading. It's more like a helper function to help you write less code... 🤷

Good point. I guess we shouldn't document it as a controller, and maybe we should even rename it to avoid confusion. Perhaps replacing "build" with "create", or renaming it something like `bindFacetToDependencies`/`unbind()`.

Reply:
> This is not a controller though, since it does not use `buildController`, does not return the same shape as a controller (for example: a `subscribe` function), even if it follows the `build` pattern, with the `engine` and `props`.
> 
> That's kind of misleading. It's more like a helper function to help you write less code... 🤷

Good point. I guess we shouldn't document it as a controller, and maybe we should even rename it to avoid confusion. Perhaps replacing "build" with "create", or renaming it something like `bindFacetToDependencies`/`unbind()`.

> Other than that, I also don't see any code in this function that makes it so it * needs * to reside in Headless 🤔 Is there something I'm missing ?
>
> Said differently, what part of this helper function could have not been copy pasted pretty much exactly as is in Atomic, and that would not have worked just as well ?

Yes, that's something I forgot to mention in this PR. Subscribing directly to the engine is something I hadn't tried, and I hadn't thought about the fact that it would be called before the url manager, my bad.

However, as mentioned in my other comment, I think we should still commit to providing some kind of helper.